### PR TITLE
Added selfhosting guide

### PR DESCRIPTION
**Pull request by hidde-j**
I made a basic guide if you want to host a server at home instead of using a VPS. The server.html still needs to be edited to link to this.

### CODE REVIEW

**Comment by LukeSmithxyz** [comment]
Great. I'll check validate and add this when I get the chance.

File: selfhosting.html [hunk 1]
```diff
@@ -50,7 +52,7 @@ <h3>Installing Debian</h3>
 While installing, do not install any desktop environment, but do install the SSH server.
 </p>
 <h3>Port forwaring</h3>
-<p>Every time you are going to use a port, you need to forward it. For example, HTTP is port 80, HTTPS is 443, etc. You need to set this up on your router's NAT settings (sometimes just port forwarding, differs per router). These steps differ for each router. Refer to your routers manual. A simple command to see what your IP address is, is <code>ifconfig</code>. This shows a lot of network info, but it will also show your local IP address needed for port forwarding.
+<p>Every time you are going to use a port, you need to forward it. For example, HTTP is port 80, HTTPS is 443, etc. You need to set this up on your router's NAT settings (sometimes just port forwarding, differs per router). These steps differ for each router. Refer to your routers manual. A simple command to see what your servers IP address is, is to run <code>ifconfig</code> on your server. This shows a lot of network info, but it will also show your local IP address needed for port forwarding.
```
Comment (DGKSK8LIFE) [inline selfhosting.html:55]: Don't like how the first sentence is worded but that's negligible.

Comment (hidde-j) [reply selfhosting.html]: I see what you mean, it should be fixed. I also fixed some other minor things.

**Review by DGKSK8LIFE — APPROVED** [review]

**Comment by uranderu** [comment]
Maybe for an update talk about the downsides of self-hosting? Ie maybe you are in breach of contract with your ISP, the security dangers of opening up your home network to the entire world, etc.

**Comment by hidde-j** [comment]
> Maybe for an update talk about the downsides of self-hosting? 

Good idea. Added some downsides. People should know what they're getting into.

**Review by DGKSK8LIFE — CHANGES_REQUESTED** [review]
I think this list makes a lot of assumptions about self-hosting that could vary from user to user.

**Comment by hidde-j** [comment]
> I think this list makes a lot of assumptions about self-hosting that could vary from user to user.
You are right, but I stated the **possible** downsides. Should I just add that they could vary from user to user?

**Review by DGKSK8LIFE — APPROVED** [review]

**Comment by binex-dsk** [comment]
> Maybe for an update talk about the downsides of self-hosting? Ie maybe you are in breach of contract with your ISP, the security dangers of opening up your home network to the entire world, etc.

Another downside is blocked ports. A LOT of ISPs block port 25, I guess to prevent SMTP spam, and some also block ports 80 and 443. Personally suffer from an ISP that blocks port 25, so I guess suggesting getting a better ISP or (for HTTP) using another port or (for SMTP) using some external relay server.

Blocked ports can be checked on https://canyouseeme.org. Very highly recommend you add this, as it not only checks blocked ports, but can verify port forwarding

A few more things:
- Split the lines up, they're way too long right now
- You don't necessarily have to use Debian. Arch is a highly viable option for servers as well--I use it for mine and it's great. It does "technically" have less software support, but the AUR nullifies this. Also, personally a lot of packages seem to be way easier to use and configure and deal with on arch
- At least somewhere it should be mentioned that most server software you'll use can pretty much run on a microwave, so even some old ancient laptop will just werk
- If you use a desktop computer on ethernet, you can very easily set that up as your server as well

v

**Comment by uranderu** [comment]
@binex-dsk Canyouseeme is not a foolproof method of checking if ports are blocked. It is very much possible that you're able to use specific ports but that they are closed by default.

Luke specified somewhere that all tutorials must be made with Debian 10 in mind. So Arch is not a possibility and IMO a weird choice for a server in general.

**Comment by hidde-j** [comment]
@binex-dsk 
> Another downside is blocked ports.
I meant to include this in the "ISP not approving on what you're doing" thing, but I see that it needed a bit more explanation.
> At least somewhere it should be mentioned that most server software you'll use can pretty much run on a microwave, so even some old ancient laptop will just werk
You are right in this regard, but I at least recommend an x86-64 bit processor for 'future proofing'. Debian is currently supported on 32 bit CPUs, but as other OSes are dropping support for them, I think Debian will so, too. Pentium 4's for example will do the job, but they are very power hungry. If you go to Pentium 2/3 or even lower, those are going to eventually struggle if you use a lot of PHP, while also doing an emailserver/fileserver.

@uranderu 
> Canyouseeme is not a foolproof method of checking if ports are blocked.
Was wondering the same thing. Doesn't it only check if the ports are correctly forwarded?
> Luke specified somewhere that all tutorials must be made with Debian 10 in mind.
Luke did, which is why I primarily said to use Debian, apart from Arch being a weird choice for a server in my opion, too.

**Comment by binex-dsk** [comment]
well yeah, any 64-bit computer would probably work

canyouseeme.org does check blocked ports. My port 25 is blocked and canyouseeme.org correctly verified this

**Comment by uranderu** [comment]
@hidde-j You are correct, youcanseeme only shows ports that are resolvable. My ISP for instance allows ports 80 and 443 to be used. But if my firewall blocks them then canyouseeme will say that they are blocked.

Imho canyouseeme is a very very unreliable way of checking if your isp block ports. I highly recommended just checking your contract or searching online for answers.

**Pull request merged**